src/applications/hca/components/FormFields/VaMedicalCenter.jsx

```diff
@@ -50,9 +89,30 @@ const VaMedicalCenter = props => {
     [facilities],
   );
 
+  useEffect(
+    () => {
+      // To prevent submissions with values missing, we need to manually blank this.
+      //   Unsetting the select's value sets it to '', but the form only blocks
+      //   submission if the value is null or undefined. Without this, the form allows
+      //   Continuing with both values set, going Back, unsetting the Facility, and then
+      //   Continuing again, resulting in a State but no Facility.
+      //   Similarly, selecting a State and Facility and then unselecting the State
+      //   allows Continuing without a State set, causing the entire Review Form
+      //   view to collapse to nothing.
+      onChange({
+        'view:facilityState': localData['view:facilityState'] || undefined,
+        vaMedicalFacility: localData.vaMedicalFacility || undefined,
+      });
+    },
+    [localData],
```

Comment:
Here, OTOH, we're directly referencing `onChange`, hence the missing dependency warning.